api/src/contexts/mail/interactors/OrderCompletedInteractor.ts

```diff
@@ -0,0 +1,37 @@
+import { IOrderCompletedInteractor } from '../usecases/IOrderCompletedInteractor';
+import { IOrderRepository } from 'src/contexts/orders/domains/repositories/IOrderRepository';
+import { IUserRepository } from '../../auth/domains/repositories/IUserRepository';
+import { IEmailAdapter } from '../domains/adapters/IEmailAdapter';
+import { ORDER_COMPLETED_TEMPLATE } from '../interactors/templates/OrderCompletedTemplate';
+
+export class OrderCompletedInteractor implements IOrderCompletedInteractor {
+  constructor(
+    private readonly emailAdapter: IEmailAdapter,
+    private readonly orderRepository: IOrderRepository,
+    private readonly userRepository: IUserRepository,
+  ) {}
+
+  async OrderCompleted(orderId: string): Promise<void> {
+    const order = await this.orderRepository.getById(Number(orderId));
```

Comment:
引数としてnumber型で渡ってくればここでのキャストは不要になりそうです。
```suggestion
    const order = await this.orderRepository.getById(orderId);
```